Orange/tests/test_classification.py

```diff
@@ -313,3 +319,28 @@ def test_adequacy_all_learners_multiclass(self):
             except TypeError as err:
                 traceback.print_exc()
                 continue
+
+class LearnerReprs(unittest.TestCase):
+    def test_reprs(self):
+        lr = LogisticRegressionLearner(tol=0.0002)
+        m = MajorityLearner()
+        nb = NaiveBayesLearner()
+        rf = RandomForestLearner(bootstrap=False, n_jobs=3)
+        st = SimpleTreeLearner(seed=1, bootstrap=True)
+        sm = SoftmaxRegressionLearner()
+        svm = SVMLearner(shrinking=False)
+        lsvm = LinearSVMLearner(tol=0.022, dual=False)
+        nsvm = NuSVMLearner(tol=0.003, cache_size=190)
+        osvm = OneClassSVMLearner(degree=2)
+        tl = TreeLearner(max_leaf_nodes=3, min_samples_split=1)
+        knn = KNNLearner(n_neighbors=4)
+        el = EllipticEnvelopeLearner(store_precision=False)
+        srf = SimpleRandomForestLearner(n_estimators=20)
+
+        learners = [lr, m, nb, rf, st, sm, svm,
+            lsvm, nsvm, osvm, tl, knn, el, srf]
+
+        for l in learners:
+            repr_str = repr(l)
+            new_l = eval(repr_str)
+            self.assertEqual(repr(new_l), repr_str)
```

Comment:
So this is basically
```py
for learner in learners:
    self.assertEqual(repr(learner), repr(eval(repr(learner))))
```
The thing this code doesn't check is equality of `repr(learner)` to the above instantiation line (e.g. `LogisticRegressionLearner(tol=0.0002)`). Could these instantiantions be in string form and eval'd first.